File: app/src/main/java/com/example/beingnotified/ui/screens/RetailerDetailScreen.kt
package com.example.beingnotified.ui.screens

import android.widget.Toast
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RetailerDetailScreen(
    retailerId: String,
    viewModel: MainViewModel,
    onNavigateBack: () -> Unit
) {
    val retailers by viewModel.retailers.collectAsStateWithLifecycle()
    val retailer = retailers.find { it.id == retailerId }
    val context = LocalContext.current
    val coroutineScope = rememberCoroutineScope()
    
    var showVoiceSheet by remember { mutableStateOf(false) }
    var isScanning by remember { mutableStateOf(false) }
    
    val cameraLauncher = rememberLauncherForActivityResult(ActivityResultContracts.TakePicturePreview()) { bitmap ->
        if (bitmap != null) {
            coroutineScope.launch {
                isScanning = true
                delay(2000) // Simulate TinyML processing
                isScanning = false
                Toast.makeText(context, "Anomaly Detected: Competitor Promotion\nPriority Score +50", Toast.LENGTH_LONG).show()
                viewModel.boostPriority(retailerId, 50)
            }
        }
    }

    if (retailer == null) return

    val nbaText = if (retailer.inventoryLevel.lowercase() == "low") {
        "Pitch Syngenta Voliam Targo (Insecticide)"
    } else if (retailer.priorityScore >= 80) {
        "Discuss Bulk Discount on Amistar Top"
    } else {
        "Conduct Routine Inventory Check"
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(retailer.name) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                }
            )
        },
        floatingActionButton = {
            FloatingActionButton(
                onClick = { showVoiceSheet = true },
                containerColor = MaterialTheme.colorScheme.secondary
            ) {
                Icon(Icons.Default.Mic, contentDescription = "Voice Assistant", tint = Color.White)
            }
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(16.dp)
        ) {
            // NBA Card
            Card(
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primary),
                shape = RoundedCornerShape(12.dp)
            ) {
                Column(modifier = Modifier.padding(20.dp).fillMaxWidth()) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Default.Lightbulb, contentDescription = null, tint = Color(0xFFFFC107))
                        Spacer(modifier = Modifier.width(8.dp))
                        Text("Next Best Action", color = Color.White, style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.Bold)
                    }
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(nbaText, color = Color.White, style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.SemiBold)
                }
            }
            
            Spacer(modifier = Modifier.height(16.dp))

            // LIME AI Reasoning
            var expanded by remember { mutableStateOf(false) }
            Card(
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Column(modifier = Modifier.fillMaxWidth()) {
                    Row(
                        modifier = Modifier.fillMaxWidth().clickable { expanded = !expanded }.padding(16.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(Icons.Default.Psychology, contentDescription = null, tint = MaterialTheme.colorScheme.secondary)
                        Spacer(modifier = Modifier.width(8.dp))
                        Text("AI Reasoning (LIME)", fontWeight = FontWeight.Bold, modifier = Modifier.weight(1f))
                        Icon(if (expanded) Icons.Default.KeyboardArrowUp else Icons.Default.KeyboardArrowDown, contentDescription = null)
                    }
                    
                    AnimatedVisibility(visible = expanded) {
                        Column(modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp)) {
                            LimeFactor("+60% Weight:", "Satellite NDVI shows crop stress in a 5km radius.", Color(0xFF4CAF50))
                            Spacer(modifier = Modifier.height(12.dp))
                            LimeFactor("+30% Weight:", "Store inventory for Voliam Targo is critically low.", Color(0xFFFF9800))
                            Spacer(modifier = Modifier.height(12.dp))
                            LimeFactor("+10% Weight:", "Routine visit overdue.", Color(0xFF2196F3))
                        }
                    }
                }
            }
            
            Spacer(modifier = Modifier.height(24.dp))
            
            Button(
                onClick = { cameraLauncher.launch(null) },
                modifier = Modifier.fillMaxWidth().height(56.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFFF9800)),
                enabled = !isScanning
            ) {
                if (isScanning) {
                    CircularProgressIndicator(color = Color.White, modifier = Modifier.size(24.dp))
                    Spacer(modifier = Modifier.width(12.dp))
                    Text("Analyzing Edge ML Model...")
                } else {
                    Icon(Icons.Default.CameraAlt, contentDescription = null)
                    Spacer(modifier = Modifier.width(12.dp))
                    Text("Anomaly Scanner (Edge Vision)")
                }
            }
        }
        
        if (showVoiceSheet) {
            VoiceBottomSheet(retailerName = retailer.name) { showVoiceSheet = false }
        }
    }
}

@Composable
fun LimeFactor(weight: String, reason: String, color: Color) {
    Row(verticalAlignment = Alignment.Top) {
        Box(
            modifier = Modifier
                .background(color.copy(alpha = 0.1f), RoundedCornerShape(4.dp))
                .border(1.dp, color.copy(alpha = 0.5f), RoundedCornerShape(4.dp))
                .padding(horizontal = 8.dp, vertical = 4.dp)
        ) {
            Text(weight, color = color, fontWeight = FontWeight.Bold, style = MaterialTheme.typography.bodySmall)
        }
        Spacer(modifier = Modifier.width(12.dp))
        Text(reason, style = MaterialTheme.typography.bodyMedium, modifier = Modifier.weight(1f))
    }
}

File: field_force_intelligence/lib/screens/retailer_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../models/retailer.dart';
import '../data/retailer_repository.dart';
import '../services/vision_service.dart';

class RetailerDetailScreen extends StatefulWidget {
  final Retailer retailer;

  const RetailerDetailScreen({Key? key, required this.retailer}) : super(key: key);

  @override
  State<RetailerDetailScreen> createState() => _RetailerDetailScreenState();
}

class _RetailerDetailScreenState extends State<RetailerDetailScreen> {
  final VisionService _visionService = VisionService();
  bool _isScanning = false;

  String _getNBA() {
    if (widget.retailer.inventoryLevel.toLowerCase() == 'low') {
      return "Pitch Syngenta Voliam Targo (Insecticide)";
    } else if (widget.retailer.priorityScore >= 80) {
      return "Discuss Bulk Discount on Amistar Top";
    }
    return "Conduct Routine Inventory Check";
  }

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.retailer.name),
        backgroundColor: theme.colorScheme.primary,
        foregroundColor: Colors.white,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            // Retailer Info
            Card(
              elevation: 2,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text("Retailer Details", style: theme.textTheme.titleMedium?.copyWith(fontWeight: FontWeight.bold)),
                    const Divider(),
                    const SizedBox(height: 8),
                    _buildInfoRow(Icons.inventory, "Inventory Level", widget.retailer.inventoryLevel),
                    const SizedBox(height: 12),
                    _buildInfoRow(Icons.score, "Priority Score", widget.retailer.priorityScore.toString()),
                    const SizedBox(height: 12),
                    _buildInfoRow(Icons.calendar_today, "Last Visited", widget.retailer.lastVisited.toLocal().toString().split(' ')[0]),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 24),
            
            // Anomaly Scanner
            ElevatedButton.icon(
              onPressed: _isScanning ? null : () => _scanAnomaly(context),
              style: ElevatedButton.styleFrom(
                backgroundColor: Colors.orange,
                foregroundColor: Colors.white,
                padding: const EdgeInsets.symmetric(vertical: 16),
                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
              ),
              icon: _isScanning 
                  ? const SizedBox(width: 20, height: 20, child: CircularProgressIndicator(color: Colors.white, strokeWidth: 2))
                  : const Icon(Icons.camera_alt),
              label: Text(_isScanning ? "Analyzing Edge ML Model..." : "Anomaly Scanner (Edge Vision)"),
            ),
            const SizedBox(height: 24),
            
            // Next Best Action (NBA) Card
            Card(
              color: theme.colorScheme.primary,
              elevation: 4,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
              child: Padding(
                padding: const EdgeInsets.all(20.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      children: [
                        const Icon(Icons.lightbulb, color: Colors.amber, size: 28),
                        const SizedBox(width: 8),
                        Text(
                          "Next Best Action",
                          style: theme.textTheme.titleLarge?.copyWith(
                            color: Colors.white,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                      ],
                    ),
                    const SizedBox(height: 16),
                    Text(
                      _getNBA(),
                      style: theme.textTheme.headlineSmall?.copyWith(
                        color: Colors.white,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 16),
            
            // Explainable AI (LIME) Expander
            Card(
              elevation: 2,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
              child: Theme(
                data: theme.copyWith(dividerColor: Colors.transparent),
                child: ExpansionTile(
                  leading: const Icon(Icons.psychology, color: Colors.green),
                  title: const Text("AI Reasoning", style: TextStyle(fontWeight: FontWeight.bold)),
                  childrenPadding: const EdgeInsets.only(left: 16, right: 16, bottom: 16),
                  children: [
                    _buildLimeFactor(context, "+60% Weight:", "Satellite NDVI shows crop stress in a 5km radius.", Colors.green),
                    const SizedBox(height: 12),
                    _buildLimeFactor(context, "+30% Weight:", "Store inventory for Voliam Targo is critically low.", Colors.orange),
                    const SizedBox(height: 12),
                    _buildLimeFactor(context, "+10% Weight:", "Routine visit overdue.", Colors.blue),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () => _showVoiceAssistant(context, widget.retailer.name),
        backgroundColor: Theme.of(context).colorScheme.secondary,
        child: const Icon(Icons.mic, color: Colors.white),
      ),
    );
  }

  Future<void> _scanAnomaly(BuildContext context) async {
    setState(() => _isScanning = true);
    
    final result = await _visionService.scanAnomaly();
    
    if (!mounted) return;
    setState(() => _isScanning = false);

    if (result != null) {
      // Show result
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text('Anomaly Detected: $result\\nPriority Score +50'),
          backgroundColor: Colors.red,
          duration: const Duration(seconds: 4),
        ),
      );

      // Update Database
      final repo = context.read<RetailerRepository>();
      await repo.boostRetailerPriority(widget.retailer.id, 50);
    }
  }

  void _showVoiceAssistant(BuildContext context, String retailerName) {
    showModalBottomSheet(
      context: context,
      shape: const RoundedRectangleBorder(borderRadius: BorderRadius.vertical(top: Radius.circular(20))),
      builder: (context) {
        return Container(
          padding: const EdgeInsets.all(24),
          height: 300,
          child: Column(
            mainAxisSize: MainAxisSize.min,
            children: [
              const Icon(Icons.graphic_eq, size: 48, color: Colors.green),
              const SizedBox(height: 16),
              const Text(
                "Listening... (Bhashini API Mock)",
                style: TextStyle(color: Colors.grey, fontStyle: FontStyle.italic),
              ),
              const SizedBox(height: 24),
              Text(
                '"Why am I visiting \$retailerName?"',
                style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 24),
              Container(
                padding: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                  color: Colors.blue.shade50,
                  borderRadius: BorderRadius.circular(8),
                  border: Border.all(color: Colors.blue.shade20),
                ),
                child: const Text(
                  "System: Satellite NDVI shows crop stress in a 5km radius, and store inventory for Voliam Targo is critically low.",
                  style: TextStyle(color: Color(0xFF0F2F6C), fontWeight: FontWeight.w600),
                ),
              ),
            ],
          ),
        );
      },
    );
  }

  Widget _buildInfoRow(IconData icon, String label, String value) {
    return Row(
      children: [
        Icon(icon, size: 20, color: Colors.grey[600]),
        const SizedBox(width: 8),
        Text("$label:", style: TextStyle(color: Colors.grey[700])),
        const SizedBox(width: 8),
        Text(value, style: const TextStyle(fontWeight: FontWeight.bold)),
      ],
    );
  }

  Widget _buildLimeFactor(BuildContext context, String weight, String reason, Color color) {
    return Row(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Container(
          margin: const EdgeInsets.only(top: 2),
          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
          decoration: BoxDecoration(
            color: color.withOpacity(0.1),
            borderRadius: BorderRadius.circular(6),
            border: Border.all(color: color.withOpacity(0.5)),
          ),
          child: Text(
            weight,
            style: TextStyle(color: color, fontWeight: FontWeight.bold, fontSize: 12),
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: Text(reason, style: const TextStyle(fontSize: 14, height: 1.4)),
        ),
      ],
    );
  }
}
